notification/cmd/main.go

```diff
@@ -0,0 +1,71 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"os/signal"
+	"syscall"
+
+	"github.com/GrabItYourself/giys-backend/lib/logger"
+	"github.com/GrabItYourself/giys-backend/lib/rabbitmq"
+	"github.com/GrabItYourself/giys-backend/lib/rabbitmq/types"
+	"github.com/GrabItYourself/giys-backend/notification/internal/config"
+	"github.com/GrabItYourself/giys-backend/notification/internal/handler"
+	"github.com/pkg/errors"
+	amqp "github.com/rabbitmq/amqp091-go"
+)
+
+func main() {
+	// Context
+	ctx, cancel := signal.NotifyContext(context.Background(), syscall.SIGTERM, syscall.SIGINT)
+	defer cancel()
+
+	// Config
+	conf := config.InitConfig()
+
+	// Logger
+	logger.InitLogger(&conf.Log)
+
+	// Connect to RabbitMQ
+	conn, err := amqp.Dial(conf.RabbitMQ.URL)
+	if err != nil {
+		logger.Fatal(errors.Wrap(err, "Failed to connect to RabbitMQ").Error())
+	}
+	defer conn.Close()
+
+	emailConsumer, err := rabbitmq.NewConsumer(conn, "email", "email-consumer")
+	if err != nil {
+		logger.Fatal(errors.Wrap(err, "Can't initialize consumer").Error())
+	}
+	defer emailConsumer.Close()
+
+	h := handler.NewHandler(&conf.EmailConfig)
+
+	done := make(chan bool)
+
+	go func() {
+		for d := range emailConsumer.Messages {
+			emailMessage := types.EmailMessage{}
+			if err := json.Unmarshal([]byte(d.Body), &emailMessage); err != nil {
+				logger.Panic(errors.Wrap(err, "Can't get email message").Error())
+			}
+
+			err := h.HandleEmailMessage(&emailMessage)
+			if err != nil {
+				logger.Panic(errors.Wrap(err, "Can't handle email message").Error())
+			}
+		}
+
+		done <- true
+	}()
```

Comment:
No need to make this a goroutine if you're gonna wait for this goroutine to finish in the main thread anyways. Just call `for d := range emailConsumer.Messages` in the main thread, and remove `done` channel.